### PR TITLE
fix: Avoid comparing None in Access-Code and Discount-Code schema validation

### LINKED ISSUE #4809: HTTP 422 in creating Access-codes
**Describe the bug**

The Server sends HTTP 422 Unprocessable entity when making a POST request like this:
``` Javascript
"attributes": {
      "code": "AC105",
      "access-url": "https://my-access-code-url.com",
      "is-active": "true",
      "tickets-number": "105",
      "min-quantity": null,
      "max-quantity": null,
      "valid-from": "2018-05-18T18:30:00+00:00",
      "valid-till": "2018-08-24T18:30:00+00:00"
    }
```
The problem is in the fact that min-quantity and the max-quantity are null which it can be when making requests from Java or JS. These are optional fields and hence these kind of errors shouldn't occur.

**To Reproduce**
1. Make a request like this.

**Expected behavior**

**Stacktrace**

```Python
127.0.0.1 - - [28/May/2018 19:25:10] "POST /v1/access-codes HTTP/1.1" 500 -
INFO:werkzeug:127.0.0.1 - - [28/May/2018 19:25:10] "POST /v1/access-codes HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/media/srv_twry/work/Projects/Community-Projects/Fossasia/open-event-server/app/__init__.py", line 64, in __call__
    return self.app(environ, start_response)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/app.py", line 2295, in wsgi_app
    response = self.handle_exception(e)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/app.py", line 1741, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/media/srv_twry/work/Projects/Community-Projects/Fossasia/open-event-server/src/flask-rest-jsonapi/flask_rest_jsonapi/decorators.py", line 32, in wrapper
    return func(*args, **kwargs)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/flask/views.py", line 88, in view
    return self.dispatch_request(*args, **kwargs)
  File "/media/srv_twry/work/Projects/Community-Projects/Fossasia/open-event-server/src/flask-rest-jsonapi/flask_rest_jsonapi/resource.py", line 68, in dispatch_request
    response = method(*args, **kwargs)
  File "/media/srv_twry/work/Projects/Community-Projects/Fossasia/open-event-server/src/flask-rest-jsonapi/flask_rest_jsonapi/decorators.py", line 56, in wrapper
    return func(*args, **kwargs)
  File "/media/srv_twry/work/Projects/Community-Projects/Fossasia/open-event-server/src/flask-rest-jsonapi/flask_rest_jsonapi/resource.py", line 182, in post
    data, errors = schema.load(json_data)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/marshmallow/schema.py", line 578, in load
    result, errors = self._do_load(data, many, partial=partial, postprocess=True)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/marshmallow_jsonapi/schema.py", line 232, in _do_load
    result = super(Schema, self)._do_load(data, many, **kwargs)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/marshmallow/schema.py", line 673, in _do_load
    field_errors=field_errors)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/marshmallow/schema.py", line 925, in _invoke_validators
    pass_original=pass_original)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python3.6/site-packages/marshmallow/marshalling.py", line 181, in run_validator
    res = validator_func(output, original_data)
  File "/media/srv_twry/work/Projects/Community-Projects/Fossasia/open-event-server/app/api/schema/access_codes.py", line 56, in validate_order_quantity
    if data['min_quantity'] >= data['max_quantity']:
TypeError: '>=' not supported between instances of 'NoneType' and 'NoneType'

```

**Found while working on** https://github.com/fossasia/open-event-frontend/pull/1104

### PR DESCRIPTION
**Pull request by schedutron**
Fixes #4809

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
If minimum and maximum quantities are not specified, they should not be compared, since that throws a `TypeError`. This PR adds this corresponding check.

### CODE REVIEW

**Comment by schedutron** [comment]
@srv-twry I've added the checks. However, I was wondering if the same could be accomplished via a `try-except` clause?

**Comment by schedutron** [comment]
@srv-twry Thanks, makes sense.

**Comment by srv-twry** [comment]
@schedutron I think ```if``` statement would be more readable assuming that you'll be catching ```TypeError```. Let's wait for the mentors to review. I would go with ```if``` wrapper only.

**Comment by srv-twry** [comment]
When you fix the hound issues, make sure you squash the commits and also change it's message like the PR title

**Comment by schedutron** [comment]
Will squash, but Hound brings a new issue with every push!

File: app/api/schema/access_codes.py [hunk 1]
```diff
@@ -53,9 +53,13 @@ def validate_order_quantity(self, data, original_data):
                 data['tickets_number'] = access_code.tickets_number
 
         if 'min_quantity' in data and 'max_quantity' in data:
-            if data['min_quantity'] >= data['max_quantity']:
-                raise UnprocessableEntity({'pointer': '/data/attributes/min-quantity'},
-                                          "min-quantity should be less than max-quantity")
+            if data['min_quantity'] is not None and \
```
Comment (srv-twry) [inline app/api/schema/access_codes.py]: Can ```'min_quantity' in data``` and ```data['min_quantity'] is not None``` be combined ?
Just asking. Not experienced in python

Comment (schedutron) [reply app/api/schema/access_codes.py]: Thought about it, we can do something like:
```
min_quantity = data.get('min_quantity', None)
max_quantity = data.get('max_quantity', None)
if min_quantity is not None and max_quantity is not None:
        if min_quantity >= max_quantity:
            raise UnprocessableEntity(
                {'pointer': '/data/attributes/min-quantity'},
                "min-quantity should be less than max-quantity"
            )
```

Comment (srv-twry) [reply app/api/schema/access_codes.py]: Yes, looks a lot better.

Comment (schedutron) [reply app/api/schema/access_codes.py]: Should I do that, or wait for mentors' review?

Comment (srv-twry) [reply app/api/schema/access_codes.py]: Let's wait
@iamareebjamal @poush @niranjan94 Please suggest if it's good.

File: app/api/schema/discount_codes.py [hunk 2]
```diff
@@ -126,9 +130,13 @@ def validate_quantity(self, data, original_data):
...
         if 'min_quantity' in data and 'max_quantity' in data:
-            if data['min_quantity'] >= data['max_quantity']:
-                raise UnprocessableEntity({'pointer': '/data/attributes/min-quantity'},
-                                          "min-quantity should be less than max-quantity")
+            if data['min_quantity'] is not None and \
+                    data['max_quantity'] is not None:
+                if data['min_quantity'] >= data['max_quantity']:
+                    raise UnprocessableEntity(
+                        {'pointer': '/data/attributes/min-quantity'},
+                        "min-quantity should be less than max-quantity"
```
[the remark this answers is not in the document]
Comment (bhaveshAn) [reply app/api/schema/discount_codes.py]: Cool!

**Comment by schedutron** [comment]
Do I need to write any additional tests for these checks?

**Comment by srv-twry** [comment]
@schedutron I don't think so.

**Comment by srv-twry** [comment]
@schedutron Please update this PR so that we can merge it. It's almost done. It is a blockage for my PR on the frontend.

File: app/api/schema/discount_codes.py [hunk 3]
```diff
@@ -41,6 +41,16 @@ class Meta:
                          schema='EventSchemaPublic',
                          type_='event')
 
+    def quantity_validation_helper(self, data):
```
[the remark this answers is not in the document]
Comment (bhaveshAn) [reply app/api/schema/discount_codes.py]: Can fix this by adding decorator `@classmethod`

[the remark this answers is not in the document]
Comment (srv-twry) [reply app/api/schema/discount_codes.py]: @schedutron Please fix this :smile:

File: app/api/schema/discount_codes.py [hunk 4]
```diff
@@ -40,6 +40,17 @@ class Meta:
                          related_view_kwargs={'discount_code_id': '<id>'},
                          schema='EventSchemaPublic',
                          type_='event')
+
```
[the remark this answers is not in the document]
Comment (srv-twry) [reply app/api/schema/discount_codes.py]: @schedutron Please remove it.

**Comment by srv-twry** [comment]
@niranjan94 Please review and merge :)
It's a blockage for another PR in frontend.

**Pull request merged**